main.go

```diff
@@ -89,12 +91,58 @@ func main() {
 		Namespace:              watchNamespace,
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		setupLog.Error(err, "unable to create manager")
 		os.Exit(1)
 	}
 
+	ctx := ctrl.SetupSignalHandler()
+
+	// Fetch ClientConfigs and create the clientCache
 	clientCache := clientcache.New(mgr.GetClient(), scheme)
 
+	cConfigs := k8ssandraiov1alpha1.ClientConfigList{}
```

Comment:
Wouldn't it be simpler to add volume mounts for the secrets? With or without the ClientConfig CRD the user has to provide the secret.

Reply:
Yes, it's simpler if you only wish to mount one secret and nothing else. I don't actually want to build kubeconfigs to always use single secret, but multiple secrets. Also, the idea here was that the same CRD would be usable in the future if we have dynamic cluster selection - one could simply add a new ClientConfig (bad name, I know) item and it could even be reconciled.

Reply:
>Yes, it's simpler if you only wish to mount one secret and nothing else.

We can mount multiple secrets with a [projected volume](https://kubernetes.io/docs/tasks/configure-pod-container/configure-projected-volume-storage/). 

>I don't actually want to build kubeconfigs to always use single secret, but multiple secrets.

It would be cumbersome to have to create a separate kubeconfig and secret for each cluster. It is easy enough to create a secret with a given kubeconfig. With that said, from a security standpoint I think it would be best if the kubeconfig file includes only those clusters that are intended to be used with the operator. We can provide a script to do this for users. Pass it the name of the contexts you want and it can create a new kubeconfig. We could even have options for renaming the contexts.

Reply:
> It would be cumbersome to have to create a separate kubeconfig and secret for each cluster.

Is it? Inside GCP it would be something like ``gcloud container clusters get-credentials cluster-name``. 

https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl#generate_kubeconfig_entry

I don't really see creating a single file with only the clusters you want any simpler than single file per cluster (which then can be pushed using kubectl create secret --from-file). Given that the user must define somehow the exported secrets instead of everything.

In any case, the the current implementation allows both single one as well as multiple secrets.

Reply:
Thanks for pointing out the `gcloud` command. I looked briefly for EKS and did not find anything similar, but hopefully there is as well as for AKS.

>In any case, the the current implementation allows both single one as well as multiple secrets.

👍

Reply:
>Is it? Inside GCP it would be something like `gcloud container clusters get-credentials cluster-name`.

@burmanm I wanted to circle back to this since I am working on some docs. I don't know why I failed to remember at the time, but that command only adds an entry in your kubeconfig file.